surface/include/pcl/surface/impl/organized_fast_mesh.hpp

```diff
@@ -102,10 +101,11 @@ pcl::OrganizedFastMesh<PointInT>::makeQuadMesh (std::vector<pcl::Vertices>& poly
   for (int y = 0; y < last_row; y += triangle_pixel_size_rows_)
   {
     // Initialize a new row
-    i = y * input_->width;
-    index_right = i + triangle_pixel_size_columns_;
-    index_down = i + y_big_incr;
-    index_down_right = i + x_big_incr;
+    int i = y * input_->width;
+    int index_right = i + triangle_pixel_size_columns_;
+    int index_down = i + y_big_incr;
+    int index_down_right = i + x_big_incr;
+    int idx = 0;
```

Comment:
Will these also be `index_t`?

Reply:
Depends on type of `input_->width`

Reply:
`std::uint32_t`. So, yes, right?

Reply:
If the code is accessing memory, then `index_t` would make better sense. If it's used in another computation, `std::uint64_t` would be better (following the rules of multiplication).

Reply:
> If the code is accessing memory, then index_t would make better sense. 

What is the reason for this?

It's point cloud's width, and at another place, is used as `index_t`. So, change to `index_t`?
https://github.com/PointCloudLibrary/pcl/blob/383ae13efd845bb18ed715b3620ef02768b9a7e8/features/include/pcl/features/impl/brisk_2d.hpp#L456

Reply:
> What is the reason for this?

We're trying to formalize the memory access guarantee offered by PCL to say that "PCL will work without fault with anything accessible by indices going from 0 to `std::numeric_limits<pcl::index_t>::max()`". This is a WIP, with slow work. Current PCL offers `std::uint32_t` width and height, but only guarantees `int` indexes which can be less than `std::numeric_limits<std::uint32_t>::max()`, and is much less than the total point cloud length (width*height) which is `std::numeric_limits<std::uint64_t>::max()`

> So, change to index_t?

Yes.

Reply:
Is f7e3723 correct?